### PR TITLE
Non-collinear projectors VASP.6.

### PR DESCRIPTION
**Pull request by dariofiosca**

### CODE REVIEW

File: python/triqs_dft_tools/converters/plovasp/proj_group.py [hunk 1]
```diff
@@ -416,8 +416,9 @@ def orthogonalize_projector_matrix(self, p_matrix):
         overlap = np.dot(p_matrix, p_matrix.conj().T)
 # Calculate [O^{-1/2}]_{m m'}
         eig, eigv = np.linalg.eigh(overlap)
-        assert np.all(eig > 0.0), ("Negative eigenvalues of the overlap matrix:"
-           "projectors are ill-defined")
+        eig = np.around(eig,10)
```
Comment (the-hampel) [inline python/triqs_dft_tools/converters/plovasp/proj_group.py]: Hi @dariofiosca ,
I identified this line as the cause for the failing test. Why introducing here a round to 10 digits? Was this for debugging purpose only? Should we put in the assert again? If I revert these three lines 419-421 back everything seems to work.

File: python/triqs_dft_tools/converters/plovasp/plotools.py [hunk 2]
```diff
@@ -85,7 +85,7 @@ def check_data_consistency(pars, el_struct):
 # Check that ions inside each shell are of the same sort
     for sh in pars.shells:
         max_ion_index = max([max(gr) for gr in sh['ions']['ion_list']])
-        assert max_ion_index < el_struct.natom, "Site index in the projected shell exceeds the number of ions in the structure"
+        assert max_ion_index <= el_struct.natom, "Site index in the projected shell exceeds the number of ions in the structure"
```
Comment (the-hampel) [inline python/triqs_dft_tools/converters/plovasp/plotools.py:88]: Can you quickly comment on this? Was this wrong previously? I thought "<" is correct as `max_ion_index` starts from 0 whereas `el_struct.natom` starts at 1?

File: python/triqs_dft_tools/converters/plovasp/proj_shell.py [hunk 3]
```diff
@@ -186,6 +176,9 @@ def extract_tmatrices(self, sh_pars):
                 matrix = raw_matrix
 
             ndim = nrow
+            print("ndim = {}".format(ndim))
```
Comment (the-hampel) [inline python/triqs_dft_tools/converters/plovasp/proj_shell.py]: Are those prints only for debug, or should we keep them now? If we keep them we can maybe add a sentence what this is?

File: python/triqs_dft_tools/converters/plovasp/proj_shell.py [hunk 4]
```diff
@@ -196,7 +189,7 @@ def extract_tmatrices(self, sh_pars):
 # If no transformation matrices are provided define a default one
         self.do_transform = False
 
-        ns_dim = 2 if self.nc_flag else 1
+        ns_dim = 1 #2 if self.nc_flag else 1
```
Comment (the-hampel) [inline python/triqs_dft_tools/converters/plovasp/proj_shell.py]: is this correct? Can we remove the commented section if we want to keep it?

File: python/triqs_dft_tools/converters/plovasp/proj_shell.py [hunk 5]
```diff
@@ -277,7 +261,8 @@ def select_projectors(self, ib_win, ib_min, ib_max):
         self.ib_win = ib_win
         self.ib_min = ib_min
         self.ib_max = ib_max
-        nb_max = ib_max - ib_min + 1
+        nb_max = ib_max - ib_min + 1    
+        print("nb_max : {}".format(nb_max))
```
Comment (the-hampel) [inline python/triqs_dft_tools/converters/plovasp/proj_shell.py]: remove debug print?

File: python/triqs_dft_tools/converters/plovasp/proj_shell.py [hunk 6]
```diff
@@ -420,7 +424,7 @@ def density_of_states(self, el_struct, emesh):
                         w_k[ik, ib, isp, io, :] = proj_k * proj_k.conj()
 
 #        eigv_ef = el_struct.eigvals[ik, ib, isp] - el_struct.efermi
-        itt = el_struct.kmesh['itet'].T.copy()
+        itt = el_struct.kmesh['itet'].T
```
Comment (the-hampel) [inline python/triqs_dft_tools/converters/plovasp/proj_shell.py]: why remove the copy? Do we want to change the `el_struct.kmesh` ?

**Review by the-hampel — CHANGES_REQUESTED** [review]
Hi @dariofiosca, 

before we can merge I marked a few more spots that we should clean up and discuss. Thank you!

File: python/triqs_dft_tools/converters/plovasp/plotools.py (hunk 2 above)
Comment (dariofiosca) [reply python/triqs_dft_tools/converters/plovasp/plotools.py:88]: I am not sure about this, maybe was a debug ... but if it works also without then should be fine.

File: python/triqs_dft_tools/converters/plovasp/proj_shell.py (hunk 3 above)
Comment (dariofiosca) [reply python/triqs_dft_tools/converters/plovasp/proj_shell.py]: I would say debug

File: python/triqs_dft_tools/converters/plovasp/proj_shell.py (hunk 4 above)
Comment (dariofiosca) [reply python/triqs_dft_tools/converters/plovasp/proj_shell.py]: You can remove the commented section, yes

File: python/triqs_dft_tools/converters/plovasp/proj_shell.py (hunk 5 above)
Comment (dariofiosca) [reply python/triqs_dft_tools/converters/plovasp/proj_shell.py]: I am not sure these were written by me and not already present in a previous version, but in any case can be removed

**Pull request merged**